pkg/storer/internal/reserve/reserve.go

```diff
@@ -152,6 +152,28 @@ func (r *Reserve) Put(ctx context.Context, chunk swarm.Chunk) error {
 			if err != nil {
 				return fmt.Errorf("failed updating stamp index: %w", err)
 			}
+		} else {
```

Comment:
this separate if and else does not work.
it could be that the "new" chunk is replacing a chunk on a stamp collision, but the "new" chunk may have already been stored in the reserve with an older timestamp (and different stamp index), which means you will end up with two copies of the same chunk in the reserve. not good.

please do not deviate from the pseudo code here  https://github.com/ethersphere/bee/compare/master...pseudocode-reserve-fix